bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Strand.java

```diff
@@ -416,6 +418,73 @@ public StrandMetadata getMetadata() {
         return metadata;
     }
 
+    protected String dumpState() {
+        StringBuilder strandInfo = new StringBuilder("\tstrand " + this.id);
+        if (this.name != null && this.getName().isPresent()) {
+            strandInfo.append(" \"").append(this.getName().get()).append("\"");
+        }
+
+        strandInfo.append(" [");
+        strandInfo.append(this.metadata.getModuleOrg()).append(".").append(this.metadata.getModuleName()).append(".")
+                .append(this.metadata.getModuleVersion()).append(":").append(this.metadata.getParentFunctionName());
+        if (this.parent != null) {
+            strandInfo.append("][").append(this.parent.getId());
+        }
+        strandInfo.append("]");
+
+        if (this.isYielded()) {
+            getInfoFromYieldedState(strandInfo);
+        } else if (this.getState().toString().equals("DONE")) {
+            strandInfo.append(" [DONE]\n\n");
+        } else {
+            strandInfo.append(" [RUNNABLE]\n\n");
+        }
+
+        return strandInfo.toString();
+    }
+
+    private void getInfoFromYieldedState(StringBuilder strandInfo) {
+        StringBuilder frameStackTrace = new StringBuilder();
+        String currState = "BLOCKED";
+        boolean noCurrState = true;
+        boolean runnableState = false;
+        String stringPrefix = "\t\tat\t";
+        try {
+            int initialFrameCount = this.frames.size();
+            int processedFrameCount = 0;
+            for (Object frame : this.frames) {
+                processedFrameCount++;
+                if (noCurrState) {
+                    currState = ((FunctionFrame) frame).getYieldStatus();
+                    if (currState == null) {
+                        runnableState = true;
+                        break;
+                    }
+                    noCurrState = false;
+                }
+                String yieldLocation = ((FunctionFrame) frame).getYieldLocation();
+                if (yieldLocation == null) {
+                    runnableState = true;
+                    break;
+                }
+                frameStackTrace.append(stringPrefix).append(yieldLocation);
+                frameStackTrace.append("\n");
+                stringPrefix = "\t\t  \t";
+            }
+            if ((initialFrameCount == 0) || (processedFrameCount != initialFrameCount)) {
+                runnableState = true;
+            }
+        } catch (ConcurrentModificationException | NullPointerException e) {
+            runnableState = true;
+        }
+        if (runnableState) {
+            strandInfo.append(" [RUNNABLE]\n\n");
```

Comment:
Shall we move the state strings like `[RUNNABLE]` to constants instead of hardcoding them?